### ISSUE TITLE
Get precise orbital parameters

### ISSUE THREAD

**Issue by sahilshahpatel** [Bone008/orbiteye#35]
It turns out that the SATCAT orbital parameters (perigee, apogee, inclination) still leave out degrees of freedom, so they can't fully specify an orbit. So we will have to turn to TLEs, which is something I was trying to avoid otherwise.

So far there are two possible sources for this data:
- https://satcat.net/ allows for easy bulk download
- https://celestrak.com/NORAD/elements/gp.php?INTDES= (specifically with no value) seems to return all rows

The satcat.net source has TLEs for 23182/50559 of the rows, and celestrak has TLEs for 23324/50559. 

If we union the datasets we get only 9 more rows than celestrak by itself, so it's probably not worth it (because incorporating a new data source could add tons of inconsistency!)

One row I looked at with no TLE was a decayed satellite. I haven't checked yet whether this is true for most or all of them, but I will try doing so next!

**Comment by sahilshahpatel** [comment]
After switching to just CelesTrak, if I filter the rows with missing TLEs so that none have operational status in NON_OP/DECAYED/UNKNOWN, I am left with just 123 rows.
> An unfortunate side effect of this filtering is that our plans for showing off space junk in the 3D view may be killed by this fact... Not sure how other visualizers deal with this!


Some of these rows seem to be due to timing mismatches of the two data download (e.g. my CelesTrak satcat data says the satellite is operational, but searching for it on the website shows it as decayed).

But there are others like `2021-052B` which are labeled as operational but have no TLE data.

Without having fully confirmed it, it does seem that the `Latest Data` column of SATCAT (which we discarded for the standardized representation) fully predicts whether or not a TLE will be available (e.g. a value of NEA=No Element Available implies that there is no TLE)

**Comment by sahilshahpatel** [comment]
Commit 9bcf159e9e5ce8034ba2ac1ffdb9f75c8850fd15 resolves this (forgot to mention). I went with CelesTrak's source and just let the `tle` field of satellites be optional. Satellites without TLEs can still be used for visualizations like the heatmap, but just not with anything current-orbit-related.

As a side note, CelesTrak doesn't have CORS requests set up for the API 😢. I tried to see if we could load directly from there which would have been a cool way to always be up-to-date automatically, but it looks like we'll just need to redownload the file occasionally.
